.github/scripts/common.py

```diff
@@ -173,6 +173,28 @@ def upload(file_to_upload, host, dest_path):
     else:
         port = "22"
 
+    # sending SFTP mkdir command to the sftp interactive mode and not batch (-b) mode
+    # as the latter would exit on any mkdir error while it is most likely
+    # the first parts of the destination is already present and thus can't be created
+    sftp_commands = "\n".join(
+        [
+            f"mkdir {part}"
+            for part in list(reversed(Path(dest_path).parents)) + [dest_path]
```

Comment:
Why don't you take advantage of the `-p` option of `mkdir`?

Reply:
This is not calling the gnu/bsd mkdir binary here, it's just the `mkdir` SFTP command.